File: algorithm/lowest/merge.go
package main

import "fmt"

func main() {
    var a = [...]int{1, 2, 3, 4, 5, 6, 7, 8, 11, 33, 55, 66}
    var b = [...]int{1, 4, 5, 8, 12, 42, 45, 51, 63, 78, 88}
    fmt.Println(a)
    fmt.Println(b)
    var result []int

    var a_len int = len(a)
    var b_len int = len(b)
    var lengths = a_len + b_len
    var a_site int = 0
    var b_site int = 0

    for i := 0; i < lengths; i ++ {
        if a_site >= a_len {
            // 这里不知道go有没有更好的合并列表的办法
            for j := b_site; j < b_len; j ++ {
                result = append(result, b[j])
            }
            break
        }
        if b_site >= b_len {
            for j := a_site; j < a_len; j ++ {
                result = append(result, a[j])
            }
            break
        }
        if a[a_site] <= b[b_site] {
            result = append(result, a[a_site])
            a_site += 1
        } else {
            result = append(result, b[b_site])
            b_site += 1
        }

    }
    fmt.Println(result)
}

File: algorithm/lowest/merge.c
#include <stdio.h>

void print_array(int array[], int len)
{
    for (int i = 0; i < len; i ++) {
        printf("%d ", array[i]);
    }
    printf("\n");
}

int main(int argc, char *argv[])
{
    int a[] = {1, 2, 3, 5, 11, 23, 33, 45, 54, 62};
    int b[] = {2, 4, 6, 8, 13, 21, 32, 44, 45, 47};
    print_array(a, 10);
    print_array(b, 10);
    int a_len = sizeof(a) / sizeof(int);
    int b_len = sizeof(b) / sizeof(int);
    int lengths = a_len + b_len;
    int result[lengths];
    printf("%d\n", lengths);
    int a_site = 0;
    int b_site = 0;
    for (int i = 0; i < lengths; i ++) {
        if (a_site >= a_len) {
            for (int j = b_site; j < b_len; j ++) {
                result[i] = b[j];
                i ++;
            }
        }
        if (b_site >= b_len) {
            for (int j = a_site; j < a_len; j ++) {
                result[i] = a[j];
                i ++;
            }
        }
        if (a[a_site] <= b[b_site]) {
            result[i] = a[a_site];
            a_site += 1;
        } else {
            result[i] = b[b_site];
            b_site += 1;
        }
    }
    print_array(result, lengths);

    return 0;
}
